### ISSUE TITLE
Improve tap2junit to accept ANSI code

### ISSUE THREAD

**Issue by BridgeAR** [nodejs/build#1756]
Currently our test suite fails badly in case the error contains ANSI code. This is pretty bad since we are starting to use more colors in core, if possible.
Stripping the ANSI is not a good idea, since it's going to hide potential errors. Is it possible to add support for that soonish?

Refs: https://ci.nodejs.org/job/node-test-commit-osx/26144/nodes=osx1011/console

```
17:11:28 + mkdir out/junit
17:11:28 + tap2junit -i test.tap -o out/junit/test.xml
17:11:30 Traceback (most recent call last):
17:11:30   File "/usr/local/bin/tap2junit", line 11, in <module>
17:11:30     load_entry_point('tap2junit==0.1.4', 'console_scripts', 'tap2junit')()
17:11:30   File "/usr/local/lib/python2.7/site-packages/tap2junit/__main__.py", line 44, in main
17:11:30     result = parse(os.path.splitext(args.input.name)[0], data)
17:11:30   File "/usr/local/lib/python2.7/site-packages/tap2junit/__main__.py", line 11, in parse
17:11:30     tap_parser.parse(data)
17:11:30   File "/usr/local/lib/python2.7/site-packages/tap2junit/tap13.py", line 124, in parse
17:11:30     self._parse(StringIO.StringIO(source))
17:11:30   File "/usr/local/lib/python2.7/site-packages/tap2junit/tap13.py", line 64, in _parse
17:11:30     self.tests[-1].yaml = yamlish.load(self.tests[-1]._yaml_buffer)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yamlish.py", line 222, in load
17:11:30     out = load('\n'.join(inobj), ignore_wrong_characters)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yamlish.py", line 203, in load
17:11:30     out = yaml.load(source, Loader=_YamlishLoader)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yaml/__init__.py", line 69, in load
17:11:30     loader = Loader(stream)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yamlish.py", line 145, in __init__
17:11:30     yaml.loader.SafeLoader.__init__(self, stream)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yaml/loader.py", line 24, in __init__
17:11:30     Reader.__init__(self, stream)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yaml/reader.py", line 74, in __init__
17:11:30     self.check_printable(stream)
17:11:30   File "/usr/local/lib/python2.7/site-packages/yaml/reader.py", line 144, in check_printable
17:11:30     'unicode', "special characters are not allowed")
17:11:30 yaml.reader.ReaderError: unacceptable character #x001b: special characters are not allowed
17:11:30   in "<unicode string>", position 65
```

**Comment by richardlau** [comment]
Source is https://github.com/jbergstroem/tap2junit, I think.

Unfortunately the last time I glanced at it it's GPL code so I'm not in a position to contribute.

**Comment by sam-github** [comment]
@jbergstroem ☝️

**Comment by refack** [comment]
I'm installing from https://github.com/refack/tap2junit.git

**Comment by sam-github** [comment]
The repo lacks a copy of the GPL, it should probably have one.

I wonder if https://github.com/dhershman1/tap-junit would work as-is with ansi?

**Comment by refack** [comment]
In any case the issue is in upstream `yamlish` - https://gitlab.com/mcepl/yamlish/tree/master

**Comment by refack** [comment]
> I wonder if https://github.com/dhershman1/tap-junit would work as-is with ansi?

AFAIR `tap-junit` doesn't support the YAML extension to TAP that's why we have `tap2junit` to bridge that gap.

**Comment by Trott** [comment]
Perhaps fixed with an updrade to tap2junit 0.1.5?
